src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1466,6 +1468,16 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  def getBucketUsage(workspaceName: WorkspaceName): Future[PerRequestMessage] = {
+    dataSource.inTransaction { dataAccess =>
+      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
+        DBIO.from(gcsDAO.getBucketUsage(workspaceContext.workspace.bucketName)) map { usage =>
+          RequestComplete(BucketUsageResponse(usage))
```

Comment:
I knew this would come up but wasn't sure how to handle it. Thanks for the suggestions.

Initially, I didn't like the "better yet"; `for` implied to me working with a collection. However, the notion of having sequential generators and an eventual yield reads very nicely... better than the `flatMap`/`map` (which actually is no better than `for` in terms of implying working with a collection).

I did have to use `DBIO.from(Future.successful(...)` to get the bucket name out of the transaction context, but that makes sense.